### PR TITLE
Add documentation for API Status Kind

### LINKED ISSUE #2974: Document Status
Status is returned in error responses from the apiserver. It's not in the swagger output. We need to figure out how to express different results in error cases from success cases.

### PR DESCRIPTION
**Pull request by saad-ali**
CC @bgrant0607

### CODE REVIEW

**Comment by nikhiljindal** [comment]
Thanks!
We also want to include the Statuses in swagger output.
Not sure if there is a separate bug for that

**Comment by saad-ali** [comment]
Let's just leave #2974 open to handle that.

File: docs/api-conventions.md [hunk 1]
```diff
@@ -239,37 +229,178 @@ Examples:
...
+  * Suggested client recovery behavior
+    * Do not retry. Fix the request.
+* `405 StatusMethodNotAllowed`
+  * Indicates that that the action the client attempted to perform on the resource was not supported by the code.
+  * Suggested client recovery behavior
+    * Do not retry. Fix the request.
+* `409 StatusConflict`
+  * Indicates that an item with the given identifier already exists and can't be created or can't be updated.
+  * Suggested client recovery behavior
+    * Do not retry. Fix the request: if creating a new item, change the identifier. If updating an item, ensure the `ResourceVersion` is up to date.
```
Comment (bgrant0607) [inline docs/api-conventions.md]: This case may require more explanation. You may also want to add a forward reference to the more detailed status response, below.

Unlike the cases above, an automatic response to 409 should typically be feasible. 

For example, when updating, usually the client would GET the object, check the client's preconditions, re-merge changes if still valid according to preconditions, and retry with the updated request (including resourceVersion).

When creating, the client would GET and compare that the fields they tried to set matched the created object. Name uniqueness ensures idempotence in the presence of failure to receive a response, including due to failure of the client. A pre-existing object should be considered ok, provided it matched what the client wanted.

File: docs/api-conventions.md [hunk 2]
```diff
@@ -239,37 +229,178 @@ Examples:
...
+* `422 StatusUnprocessableEntity`
+  * Indicates that the requested create or update operation cannot be completed due to invalid data provided as part of the request.
+  * Suggested client recovery behavior
+    * Do not retry. Fix the request.
+* `429 StatusTooManyRequests`
+  * Indicates that the server has recieved more requests then it can process.
+  * Suggested client recovery behavior:
+    * Wait and retry with exponential backoff.
```
Comment (bgrant0607) [inline docs/api-conventions.md]: Actually the server provides a Retry-after header.

File: docs/api-conventions.md [hunk 3]
```diff
@@ -239,37 +229,178 @@ Examples:
...
+* `409 StatusConflict`
+  * Indicates that an item with the given identifier already exists and can't be created or can't be updated.
+  * Suggested client recovery behavior
+    * Do not retry. Fix the request: if creating a new item, change the identifier. If updating an item, ensure the `ResourceVersion` is up to date.
+* `422 StatusUnprocessableEntity`
+  * Indicates that the requested create or update operation cannot be completed due to invalid data provided as part of the request.
+  * Suggested client recovery behavior
+    * Do not retry. Fix the request.
+* `429 StatusTooManyRequests`
+  * Indicates that the server has recieved more requests then it can process.
```
Comment (bgrant0607) [inline docs/api-conventions.md]: Or that that particular client's rate limit has been exceeded.

**Comment by bgrant0607** [comment]
This is awesome! Just a couple comments.

**Comment by saad-ali** [comment]
Thanks Brian! Addressed comments.
PTAL (updated "409 StatusConflict" and "429 StatusTooManyRequests" sections)

File: docs/api-conventions.md [hunk 4]
```diff
@@ -239,37 +229,182 @@ Examples:
...
+  *   * Either change the identifier and try again, or GET and compare the fields in the pre-existing object and issue a PUT/update to modify the existing object.
+  * * If updating an existing resource:
+      * See `Conflict` from the `status` response section below on how to retrieve more information about the nature of the conflict.
+      * GET and compare the fields in the pre-existing object, merge changes (if still valid according to preconditions), and retry with the updated request (including `ResourceVersion`).
+* `422 StatusUnprocessableEntity`
+  * Indicates that the requested create or update operation cannot be completed due to invalid data provided as part of the request.
+  * Suggested client recovery behavior
+    * Do not retry. Fix the request.
+* `429 StatusTooManyRequests`
+  * Indicates that the either the client rate limit has been exceeded or the server has recieved more requests then it can process.
```
Comment (bgrant0607) [inline docs/api-conventions.md]: typo: received

**Comment by bgrant0607** [comment]
LGTM, other than the typo, which can be fixed in another PR. Thanks!

**Pull request merged**